docs/security-best-practices.md

```diff
@@ -153,6 +153,108 @@ To change the default endpoint to be `localhost`-bound in all components, enable
 
 If `localhost` resolves to a different IP due to your DNS settings then explicitly use the loopback IP instead: `127.0.0.1` for IPv4 or `::1` for IPv6. In IPv6 setups, ensure your system supports both IPv4 and IPv6 loopback addresses to avoid issues.
 
+Using `localhost` may not work in environments like Docker, Kubernetes, and other environments that have non-standard networking setups.  We've documented a few working example setups for the OTLP receiver gRPC endpoint below, but other receivers and other Collector components may need similar configuration.
+
+#### Docker
+You can run the Collector in Docker using a configuration like the following:
+
+OTEL Collector config file:
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: my-hostname:4317 # the same hostname from your docker run command
+```
+Docker run command:
+`docker run --hostname my-hostname --name container-name -p 127.0.0.1:4567:4317 -v $(pwd):/etc/otelcol otel/opentelemetry-collector:0.104.0 --config /etc/otelcol/config.yaml`
+
+You could access it from outside that Docker network (for example on a regular program running on the host) by connecting to `127.0.0.1:4567`.
+
+#### Docker Compose
+```yaml
+services:
+  otel-collector:
+    image: otel/opentelemetry-collector-contrib:0.104.0
+    ports:
+      - "4567:4317"
+```
+
+otel collector config file:
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+         endpoint: otel-collector:4317 # Using the service name from your Docker compose file
+```
+
+You can connect to this Collector from another Docker container running in the same network as the Collector by connecting to `otel-collector01:4317`.  You could access it from outside that Docker network (for example on a regular program running on the host) by connecting to `127.0.0.1:4567`.
+
+#### Kubernetes
+If you run the Collector as a `Daemonset`, you can use a configuration like below:
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: collector
+spec:
+  selector:
+    matchLabels:
+      name: collector
+  template:
+    metadata:
+      labels:
+        name: collector
```

Comment:
What's the use of this?

Reply:
[Kubernetes Deployment docs](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector
)
Just part of the Deployment manifest.  Defines the pods that a deployment will create and manage.

Reply:
ah, sorry, misread the config, yeah that makes sense lol